cdap-api/src/main/java/io/cdap/cdap/api/metadata/MetadataEntityCodec.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright © 2022 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.api.metadata;
+
+import com.google.gson.JsonDeserializationContext;
+import com.google.gson.JsonDeserializer;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParseException;
+import com.google.gson.JsonSerializationContext;
+import com.google.gson.JsonSerializer;
+import io.cdap.cdap.api.metadata.MetadataEntity.KeyValue;
+import io.cdap.cdap.internal.guava.reflect.TypeToken;
+
+import java.lang.reflect.Type;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * JSON Type adapter for MetadataEntity. Version for {@link MetadataEntityCodec#versionlessEntityType}
+ * will be removed during serialization since it is not persisted in storage and will always be the
+ * default value "-SNAPSHOT".
+ */
+public class MetadataEntityCodec implements JsonSerializer<MetadataEntity>, JsonDeserializer<MetadataEntity> {
+
+  private static final Type MAP_DETAILS_TYPE = new TypeToken<LinkedHashMap<String, String>>() { }.getType();
+  private static final Type STRING_TYPE_TYPE = new TypeToken<String>() { }.getType();
+  private static final Set<String> versionlessEntityType = new HashSet<>(Arrays.asList("application",
```

Comment:
use capital letter for static final variable

Reply:
this codec is a utility class that should be in cdap-common instead of in cdap-api, you can then use `MetadataUtil.VERSIONED_ENTITY_TYPES` instead of defining your own variable